### ISSUE TITLE
Autodiff error using clamp() function

### ISSUE THREAD

**Issue by 111116** [LuisaGroup/LuisaCompute#203]
**Describe the bug**
Differentiating `clamp(x, 0.0, 1.0)` w.r.t. `x=1.5`  yields gradient of 1, whereas the gradient is expected to be 0.

**To Reproduce
```python
from luisa import *
from luisa.autodiff import requires_grad, autodiff, backward, grad


@func
def clampa(x, a, b):
    return min(max(x, a), b)


@func
def f(x) -> float:
    return clampa(x, 0., 1.)


@func
def df(x) -> float:
    with autodiff():
        requires_grad(x)
        t = f(x)
        backward(t, 1.0)
        g = grad(x)
    return g

@func
def test_autodiff(x):
    print("testing autodiff (", x, ")")
    eps = 1e-3
    print("ad f(a) =", df(x))
    print("fd f(a) =", (f(x + eps) - f(x - eps)) / (2.0 * eps))


init()
test_autodiff(1.5, dispatch_size=1)
```

**Expected behavior**
FD and AD gradient matches.

**Additional context**
Suspecting bug: may need to change min into max
https://github.com/LuisaGroup/LuisaCompute/blob/b86ebf79fc90c11dab5396aeea1380c8bd177e08/src/rust/luisa_compute_ir/src/transform/autodiff.rs#L1234
